frost-core/src/benches.rs

```diff
@@ -0,0 +1,201 @@
+//! Ciphersuite-generic benchmark functions.
+
+use std::collections::HashMap;
+
+use criterion::{BenchmarkId, Criterion, Throughput};
+use rand_core::{CryptoRng, RngCore};
+
+use crate::{batch, frost, Ciphersuite, Signature, SigningKey, VerifyingKey};
+
+struct Item<C: Ciphersuite> {
+    vk: VerifyingKey<C>,
+    sig: Signature<C>,
+}
+
+fn sigs_with_distinct_keys<C: Ciphersuite, R: RngCore + CryptoRng + Clone>(
+    rng: &mut R,
+) -> impl Iterator<Item = Item<C>> {
+    let mut rng = rng.clone();
+    std::iter::repeat_with(move || {
+        let msg = b"Bench";
+        let sk = SigningKey::new(&mut rng);
+        let vk = VerifyingKey::from(&sk);
+        let sig = sk.sign(&mut rng, &msg[..]);
+        Item { vk, sig }
+    })
+}
+
+/// Benchmark batched signature verification with the specified ciphersuite.
+pub fn bench_batch_verify<C: Ciphersuite, R: RngCore + CryptoRng + Clone>(
+    c: &mut Criterion,
+    name: &str,
+    rng: &mut R,
+) {
+    let mut group = c.benchmark_group(format!("Batch Verification {}", name));
+    for &n in [8usize, 16, 24, 32, 40, 48, 56, 64].iter() {
+        group.throughput(Throughput::Elements(n as u64));
+
```

Comment:
###### Optional
The message `msg` could be defined here.

Reply:
See above